program-runtime/src/loaded_programs.rs

```diff
@@ -1061,16 +1074,19 @@ impl<FG: ForkGraph> ProgramCache<FG> {
                     .slot_versions
                     .iter()
                     .filter_map(move |program| match program.program {
-                        LoadedProgramType::LegacyV0(_) | LoadedProgramType::LegacyV1(_)
-                            if include_program_runtime_v1 =>
-                        {
-                            Some((*id, program.clone()))
-                        }
-                        LoadedProgramType::Typed(_) if include_program_runtime_v2 => {
-                            Some((*id, program.clone()))
+                        LoadedProgramType::Loaded(_) => {
+                            let include =
+                                if let LoadedProgramOwner::LoaderV4 = program.account_owner {
```

Comment:
nit: if you implement Eq for LoadedProgramOwner then I think this can become one
if with LoaderV4 && include_program_runtime_v2 { .. } else if
include_program_runtime_v1 { ... }